src/common/types/uuid.cpp

```diff
@@ -0,0 +1,69 @@
+#include "duckdb/common/types/uuid.hpp"
+
+#include <sstream>
+#include <iomanip>
+#include <inttypes.h>
+
+namespace duckdb {
+
+bool UUID::FromString(string str, hugeint_t &result) {
+	auto hex2char = [](char ch) -> unsigned char {
+		if (ch >= '0' && ch <= '9') {
+			return ch - '0';
+		}
+		if (ch >= 'a' && ch <= 'f') {
+			return 10 + ch - 'a';
+		}
+		if (ch >= 'A' && ch <= 'F') {
+			return 10 + ch - 'A';
+		}
+		return 0;
+	};
+	auto is_hex = [](char ch) -> bool {
+		return (ch >= '0' && ch <= '9') || (ch >= 'a' && ch <= 'f') || (ch >= 'A' && ch <= 'F');
+	};
+
+	if (str.empty()) {
+		return false;
+	}
+	int has_braces = 0;
+	if (str.front() == '{') {
+		has_braces = 1;
+	}
+	if (has_braces && str.back() != '}') {
+		return false;
+	}
+
+	result.lower = 0;
+	result.upper = 0;
+	size_t count = 0;
+	for (size_t i = has_braces; i < str.size() - has_braces; ++i) {
+		if (str[i] == '-') {
+			continue;
+		}
+		if (count >= 32 || !is_hex(str[i])) {
+			return false;
+		}
+		if (count >= 16) {
+			result.lower = (result.lower << 4) | hex2char(str[i]);
+		} else {
+			result.upper = (result.upper << 4) | hex2char(str[i]);
```

Comment:
@Mytherin About big endian could you explain further? I think current logic is right?

I also checked the golang implementation, should be the same https://github.com/google/uuid/blob/master/uuid.go#L84-L89

Reply:
For clickhouse, they use UInt128 which fine. Our problem is if the UUID start with a-f, hugeint_t will be negative.

https://github.com/ClickHouse/ClickHouse/blob/066d02dd2f/src/Core/UUID.cpp

Reply:
Yes, the signedness indeed makes this more difficult. We would need to map e.g.

`00000000-0000-0000-0000-000000000000` -> `-2^127`
`00000000-0000-0000-0000-000000000001` -> `-2^127+1`
etc for the ordering to work correctly. This is possible, but means the string <-> uuid conversions are a bit more involved as they are no longer only bit-shifting. 

Alternatively we could introduce a `PhysicalType::UINT128` (and potentially corresponding `LogicalType::UHUGEINT`). The type could be taken from [here](https://github.com/calccrypto/uint128_t/blob/master/uint128_t.cpp) and adapted to work in DuckDB.